kernel/ps2_controller/src/lib.rs

```diff
@@ -0,0 +1,99 @@
+#![no_std]
+
+use log::{debug, warn};
+use ps2::{write_command, flush_output_buffer, read_config, write_config, read_controller_test_result, read_port_test_result, PortTestResult, HostToControllerCommand::*, reset_keyboard, reset_mouse};
+
+
+// see https://wiki.osdev.org/%228042%22_PS/2_Controller#Initialising_the_PS.2F2_Controller
+/// initialize the first and second (if it exists) PS/2 port
+pub fn init() -> Result<(), &'static str> {
+    // Step 1: Initialise USB Controllers
+    // no USB support yet
+
+    // Step 2: Determine if the PS/2 Controller Exists
```

Comment:
so if there isn't a `FADT` table, is the general advice just to continue trying to init ps2, kbd, and mouse?

Reply:
"if the bit is set, or the system doesn't support ACPI (no ACPI tables and no FADT) then there is a PS/2 Controller"
What happens with "ACPI yes" and "FADT no"?
Looking at our acpi code, we currently always assume the system to support ACPI, it otherwise shuts down.
The FADT is mandatory (or our acpi code says so).

If we were to support systems without ACPI, I assume `get_acpi_tables` would return an empty table and `Fadt::get` would return `None`. I'm adding a comment.